### DISCUSSION TITLE
Add tags anywhere?

### DISCUSSION THREAD

**Discussion by yves-chevallier** [squidfunk/mkdocs-material#7394] [Ideas]
### Context

I am struggling to understand how tags work. I eventually noticed that:

- Tags will always be displayed at the top of the page.
- The displayed tags will reflect those listed in the meta data.
- An hashtag symbol is automatically prepend to the tag name if defined in `extra/tags/key:value`
- It is not possible do declare tags anywhere on a page.
- It is not possible to show tags labels for the current section (heading)

### Allows for displaying hashtags bubbles anywhere on the page

Currently tags only displays at the top of a page. When a page has plenty of tags associated, I could be preferable to display them at the end of the page, for for say anywhere the user wants. 

We could add a config that can be put either in the meta-data or in the config file: 

```yml
tags:
  labels: top|bottom
```

To put them anywhere, why not using

```html
<!-- material/hashtags -->
```

## Allows for declaring inline tags anywhere in a document

My use-case was to use tags to enhance key concepts explained in each individual sections of my documentation. Therefore, tags should be declared at the top of a section, or anywhere in a section. 

```markdown
## H2G2

The book "The Hitchhiker's Guide to the Galaxy" by Douglas Adams is a science fiction comedy series that has been adapted into various formats, including a radio series, a TV series, a movie, and a video game. The story follows the adventures of Arthur Dent, an ordinary human who is rescued from Earth just before it is destroyed to make way for a hyperspace bypass. 

<!-- tag:mouse -->
<!-- tag:42 -->
<!-- tag:sf -->
```

Then it would be nice to also show hashtags labels at the top or at the end of sections where they are declared: 

```yml
tags: 
  hashtags:
    scope-level: 1  # Put hashtags labels per heading h1
```

### Related links

- https://mkdocs-badges.six-two.dev/badges/
- https://github.com/squidfunk/mkdocs-material/discussions/6286
- https://github.com/squidfunk/mkdocs-material/discussions/6724

### Use Cases

My documentation is a programming course. It has a lot of pages and programming concepts such as `pointers`, `oop`, `graph`, `algorithm`... can be used to quickly search relevant part in the documentation that treat these concepts. 

Having tags only declared for a whole `.md` file isn't narrow enough. 

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material/), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Thanks for asking. I'm moving this to discussions, since this is more of a question than a concrete feature request and is not inline how tags are currently implemented. Tagging @alexvoss for visibility.

**Comment by Aetherinox** [comment]
Was looking into this, as being able to place tags next to a specific element  would be great. And a way to index where a tag has been placed, and a "index" which allows you to view a brief description of each area the tag is referenced in.
